### PR TITLE
feat: override subquery/result limits

### PR DESCRIPTION
**Pull request by rpalakkal**

### CODE REVIEW

File: circuit/js/src/cliHandler/compile.ts [hunk 1]
```diff
@@ -34,6 +35,8 @@ export const compile = async (
         shouldTime: options.stats,
         inputSchema: f.inputSchema,
         results: cache,
+        capacity: f.config?.capacity,
+        config: f.config?.circuit,
```
Comment (ytham) [inline circuit/js/src/cliHandler/compile.ts:39]: do we need to pass capacity and config into the `prove` cliHandler as well?

Comment (ytham) [reply circuit/js/src/cliHandler/compile.ts:39]: also is `config: f.config?.circuit` correct here?

Comment (ytham) [reply circuit/js/src/cliHandler/compile.ts:39]: discussed w/ Roshan offline; we can rename the `circuit` field in `AxiomV2CircuitConfig` to `config` to keep w/ convention

**Review by ytham — APPROVED** [review]
LGTM. note to anyone reading that the same-repo links are ok for this repo, as the publish-all script will handle this

**Comment by jonathanpwang** [comment]
@rpalakkal Need to fix merge conflicts because we moved `circuit/js` to `circuit`

File: circuit/js/src/constants.ts [hunk 2]
```diff
@@ -5,11 +5,12 @@ export const MAX_USER_OUTPUTS = AxiomV2CircuitConstant.UserMaxOutputs;
 export const MAX_DATA_SUBQUERIES = AxiomV2CircuitConstant.UserMaxSubqueries;
 export const MAX_SUBQUERY_INPUTS = AxiomV2CircuitConstant.MaxSubqueryInputs;
 export const MAX_SUBQUERY_OUTPUTS = AxiomV2CircuitConstant.MaxSubqueryOutputs;
+export const SUBQUERY_FE = (1 + MAX_SUBQUERY_INPUTS + MAX_SUBQUERY_OUTPUTS);
 
 export const MAX_SOLIDITY_MAPPING_KEYS = AxiomV2CircuitConstant.MaxSolidityMappingKeys;
 
-export const USER_COMPUTE_NUM_INSTANCES = USER_OUTPUT_FE * MAX_USER_OUTPUTS;
-export const SUBQUERY_NUM_INSTANCES = MAX_DATA_SUBQUERIES * (1 + MAX_SUBQUERY_INPUTS + MAX_SUBQUERY_OUTPUTS);
+export const USER_COMPUTE_NUM_INSTANCES = MAX_USER_OUTPUTS * USER_OUTPUT_FE;
```
Comment (jonathanpwang) [inline circuit/js/src/constants.ts:12]: is this even used anymore?

**Review by jonathanpwang — APPROVED** [review]
LGTM, please add a unit test where the override capacity is used.

**Comment by ytham** [comment]
@rpalakkal is this still WIP or ready to merge?

**Pull request merged**